pallets/funding/src/functions/5_funding_end.rs

```diff
@@ -189,4 +189,30 @@ impl<T: Config> Pallet<T> {
 			pays_fee: Pays::Yes,
 		})
 	}
+
+	pub fn finalize_funding(
+		project_id: ProjectId,
+		mut project_details: ProjectDetailsOf<T>,
+		outcome: ProjectOutcome,
+		settlement_delta: BlockNumberFor<T>,
+	) -> Result<u32, DispatchError> {
+		let now = <frame_system::Pallet<T>>::block_number();
+
+		project_details.status = match outcome {
+			ProjectOutcome::FundingSuccessful | ProjectOutcome::FundingAccepted => ProjectStatus::FundingSuccessful,
+			_ => ProjectStatus::FundingFailed,
+		};
+		ProjectsDetails::<T>::insert(project_id, project_details);
+
+		let insertion_iterations =
+			match Self::add_to_update_store(now + settlement_delta, (&project_id, UpdateType::StartSettlement)) {
+				Ok(iterations) => iterations,
+				Err(_iterations) => return Err(Error::<T>::TooManyInsertionAttempts.into()),
+			};
```

Comment:
nit: If you want to avoid the Err + into(), this should work

```
		let insertion_iterations =
			Self::add_to_update_store(now + settlement_delta, (&project_id, UpdateType::StartSettlement))
				.map_err(|_| Error::<T>::TooManyInsertionAttempts)?;
```